src/installer/tests/HostActivation.Tests/MultiArchInstallLocation.cs

```diff
@@ -71,6 +71,24 @@ public void EnvironmentVariable_ArchSpecificDotnetRootIsUsedOverDotnetRoot()
                 .And.NotHaveStdErrContaining("Using environment variable DOTNET_ROOT=");
         }
 
+        [Fact]
+        public void EnvironmentVariable_DotnetRootPathDoesNotExist()
+        {
+            var fixture = sharedTestState.PortableAppFixture
+                .Copy();
+
+            var appExe = fixture.TestProject.AppExe;
+            Command.Create(appExe)
+                .EnableTracingAndCaptureOutputs()
+                .DotNetRoot("non_existent_path")
+                .EnvironmentVariable("DOTNET_MULTILEVEL_LOOKUP", "0")
+                .Execute()
+                .Should().Pass()
```

Comment:
This will only pass if there is an existing global installation, which we don't (and shouldn't) require for building the repo and running tests.

You should be able to also add the registered install location override and/or set the environment variable for overriding the default install path. I think that would cover this case where the host should fall back to those.